AnalysisType: rule
Filename: duo_user_anomalous_push.py
RuleID: DUO.User.Denied.AnomalousPush
DisplayName: DUO User Auth Denied For Anomalous Push
Enabled: true
DedupPeriodMinutes: 15
LogTypes:
  - Duo.Authentication
Tags:
  - DUO
Severity: Medium
Description: >
  A DUO authentication was denied due to an anomalous 2FA push.
Reference: https://duo.com/docs/adminapi#authentication-logs
Runbook: Follow up with the user to confirm they intended several pushes in quick succession.
Tests:
  -
    Name: anomalous_push_occurred
    ExpectedResult: true
    Log:
      {
        "access_device": {
          "ip": "12.12.112.25",
          "os": "Mac OS X"
        },
        "auth_device": {
          "ip": "12.12.12.12"
        },
        "application": {
          "key": "D12345",
          "name": "Slack"
        },
        "event_type": "authentication",
        "factor": "duo_push",
        "reason": "anomalous_push",
        "result": "denied",
        "user": {
          "name": "example@example.io"
        }
      }
